src/commands/src/helpers/dependency-helpers.ts

```diff
@@ -0,0 +1,60 @@
+import path from 'path';
+import fs from 'fs';
+import { executeCliCommand, getCommitBranch, getCommitHash, getPatchId, getRepoDir } from '.';
+import { promisify } from 'util';
+
+const existsAsync = promisify(fs.exists);
+const writeFileAsync = promisify(fs.writeFile);
+
+async function cloneRepo(repoName: string) {
+  await executeCliCommand({
+    command: 'git',
+    args: ['clone', `https://github.com/mongodb/${repoName}`],
+    options: { cwd: `${process.cwd()}/repos` },
+  });
+}
+async function createEnvProdFile(repoDir: string, projectName: string, baseUrl: string, prefix = '') {
+  const prodFileName = `${process.cwd()}/snooty/.env.production`;
+
+  try {
+    await writeFileAsync(
+      prodFileName,
+      `GATSBY_SITE=${projectName}
+      GATSBY_MANIFEST_PATH=${repoDir}/bundle.zip
+      GATSBY_PARSER_USER=${process.env.USER}
+      GATSBY_BASE_URL=${baseUrl}
+      PATH_PREFIX=${prefix}`,
+      'utf8'
+    );
+  } catch (e) {
+    console.error(`ERROR! Could not write to .env.production`);
+    throw e;
+  }
+}
+
+export async function prepareBuildAndGetDependencies(repoName: string, projectName: string, baseUrl: string) {
+  // before we get build dependencies, we need to clone the repo
+  await cloneRepo(repoName);
+
+  const repoDir = getRepoDir(repoName);
+
+  // doing these in parallel
+  const commandPromises = [
+    getCommitHash(repoDir),
+    getCommitBranch(repoDir),
+    getPatchId(repoDir),
+    existsAsync(path.join(process.cwd(), 'config/redirects')),
+    createEnvProdFile(repoDir, projectName, baseUrl),
+  ];
+
+  const dependencies = await Promise.all(commandPromises);
```

Comment:
@branberry would we want to provide a way of catching or handling the error message if one of these promises are rejected.

Reply:
That's a good question, that is something I have been thinking about. I was thinking that maybe the error handling could occur in the place where this function gets called to simplify this code, but I am not set in stone on that.

Reply:
@branberry this was a pretty good [read](https://kinsta.com/blog/errors-in-javascript/#2-do-not-swallow-any-exceptions), on how and where to raise an error to be handled.